pennylane/tape/qscript.py

```diff
@@ -381,6 +389,15 @@ def shots(self) -> Shots:
         """
         return self._shots
 
+    @property
+    def num_preps(self) -> int:
```

Comment:
Given this will almost always be 0 or 1, I'm not concerned with the performance of this.